### PR TITLE
Refactor lemmatizer enclitics

### PR DESCRIPTION
**Pull request by Tanvez**
This PR adds the following:
- A preprocessor class to `base.py` and `lemmatizer.py` to further parse out enclitics from an already tokenized words in a latin text. Note: Placeholder methods are created for other languages(chinese, russian, and greek)
- Test for the latin preprocessor
- Changes were made to the front-end to account for words lookup_form that includes macrons

### CODE REVIEW

File: hedera/supported_languages.py [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+)
 
 
 # using ISO 639.2 CODES
 
-SupportedLanguage = namedtuple("SupportedLanguage", ["code", "verbose_name", "service", "tokenizer"])
-
+SupportedLanguage = namedtuple("SupportedLanguage", ["code", "verbose_name", "service", "tokenizer", "preprocessor"])
 
+# TODO add preprocessor to each language - ex latin enclitics lookup
```
Comment (arthurian) [inline hedera/supported_languages.py:29]: Looks like this TODO can be removed

**Review by arthurian — APPROVED** [review]
Looks good @Tanvez! Great work :+1:

**Review by jaguillette — APPROVED** [review]
LGTM!

**Comment by bilbe** [comment]
looks great

**Review by bilbe — COMMENTED** [review]
looks great!

**Pull request merged**